contracts/AssetPool.sol

```diff
@@ -64,11 +96,122 @@ contract AssetPool is Ownable {
         _deposit(msg.sender, amount);
     }
 
-    /// @notice Withdraws collateral from the asset pool. Accepts the amount of
-    ///         underwriter tokens representing the share of the pool that
-    ///         should be withdrawn.
+    /// @notice Initiates the withdrawal of collateral and rewards from the pool.
+    ///         Accepts the amount of underwriter tokens representing the share
+    ///         of the pool that should be withdrawn.
+    ///         Underwriter needs to complete the withdrawal by calling the
+    ///         `completeWithdrawal` function after the withdrawal delay passes
+    ///         but before the graceful withdrawal timeout ends to avoid part of
+    ///         their share being seized by the pool.
     /// @dev Before calling this function, underwriter token needs to have the
     ///      required amount accepted to transfer to the asset pool.
+    function initiateWithdrawal(uint256 covAmount) external {
+        uint256 covBalance = underwriterToken.balanceOf(msg.sender);
+        require(
+            covAmount <= covBalance,
+            "Underwriter token amount exceeds balance"
+        );
+        require(
+            covAmount > 0,
+            "Underwriter token amount must be greater than 0"
+        );
+
+        underwriterToken.safeTransferFrom(msg.sender, address(this), covAmount);
+
+        pendingWithdrawal[msg.sender] = covAmount.add(
+            pendingWithdrawal[msg.sender]
+        );
+        /* solhint-disable not-rely-on-time */
+        withdrawalInitiatedTimestamp[msg.sender] = block.timestamp;
+
+        emit WithdrawalInitiated(
+            msg.sender,
+            pendingWithdrawal[msg.sender],
+            block.timestamp
+        );
+        /* solhint-enable not-rely-on-time */
```

Comment:
Do we need this guy?

Reply:
I think yes, at least that's how we do it in all other places. Honestly, given how many times we need to ignore this detector, I am thinking about ignoring it in the global rules...